core/src/main/java/org/apache/carbondata/core/datastore/compression/GzipCompressor.java

```diff
@@ -0,0 +1,201 @@
+    short[] shorts = new short[unCompArray.length / ByteUtil.SIZEOF_SHORT];
+    unCompBuffer.get(shorts);
+    return shorts;
+  }
+
+  @Override public byte[] compressInt(int[] unCompInput) {
+    ByteBuffer unCompBuffer = ByteBuffer.allocate(unCompInput.length * ByteUtil.SIZEOF_INT);
+    unCompBuffer.asIntBuffer().put(unCompInput);
+    return compressData(unCompBuffer.array());
+  }
+
+  @Override public int[] unCompressInt(byte[] compInput, int offset, int length) {
+    byte[] unCompArray = unCompressByte(compInput, offset, length);
+    IntBuffer unCompBuffer = ByteBuffer.wrap(unCompArray).asIntBuffer();
+    int[] ints = new int[unCompArray.length / ByteUtil.SIZEOF_INT];
+    unCompBuffer.get(ints);
+    return ints;
+  }
+
+  @Override public byte[] compressLong(long[] unCompInput) {
+    ByteBuffer unCompBuffer = ByteBuffer.allocate(unCompInput.length * ByteUtil.SIZEOF_LONG);
+    unCompBuffer.asLongBuffer().put(unCompInput);
+    return compressData(unCompBuffer.array());
+  }
+
+  @Override public long[] unCompressLong(byte[] compInput, int offset, int length) {
+    byte[] unCompArray = unCompressByte(compInput, offset, length);
+    LongBuffer unCompBuffer = ByteBuffer.wrap(unCompArray).asLongBuffer();
+    long[] longs = new long[unCompArray.length / ByteUtil.SIZEOF_LONG];
+    unCompBuffer.get(longs);
+    return longs;
+  }
+
+  @Override public byte[] compressFloat(float[] unCompInput) {
+    ByteBuffer unCompBuffer = ByteBuffer.allocate(unCompInput.length * ByteUtil.SIZEOF_FLOAT);
+    unCompBuffer.asFloatBuffer().put(unCompInput);
+    return compressData(unCompBuffer.array());
+  }
+
+  @Override public float[] unCompressFloat(byte[] compInput, int offset, int length) {
+    byte[] unCompArray = unCompressByte(compInput, offset, length);
+    FloatBuffer unCompBuffer = ByteBuffer.wrap(unCompArray).asFloatBuffer();
+    float[] floats = new float[unCompArray.length / ByteUtil.SIZEOF_FLOAT];
+    unCompBuffer.get(floats);
+    return floats;
+  }
+
+  @Override public byte[] compressDouble(double[] unCompInput) {
+    ByteBuffer unCompBuffer = ByteBuffer.allocate(unCompInput.length * ByteUtil.SIZEOF_DOUBLE);
+    unCompBuffer.asDoubleBuffer().put(unCompInput);
+    return compressData(unCompBuffer.array());
+  }
+
+  @Override public double[] unCompressDouble(byte[] compInput, int offset, int length) {
+    byte[] unCompArray = unCompressByte(compInput, offset, length);
+    DoubleBuffer unCompBuffer = ByteBuffer.wrap(unCompArray).asDoubleBuffer();
+    double[] doubles = new double[unCompArray.length / ByteUtil.SIZEOF_DOUBLE];
+    unCompBuffer.get(doubles);
+    return doubles;
+  }
+
+  @Override public long rawCompress(long inputAddress, int inputSize, long outputAddress)
+      throws IOException {
+    throw new RuntimeException("Not implemented rawUncompress for gzip yet");
+  }
+
+  @Override public long rawUncompress(byte[] input, byte[] output) throws IOException {
+    //gzip api doesnt have rawCompress yet.
```

Comment:
if it is so, just throw exception, otherwise JVM may crash if you pass the illegal address/length

Reply:
Done.